integrations/http/pattern.go

```diff
@@ -0,0 +1,64 @@
+package http
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+)
+
+func extractKeys(s string) (keys []string, err error) {
```

Comment:
1. Missing documentation, so it's unclear to me what is expected input and what isn't valid / supposed to be possible
2. Because of that, I'm not sure if this loop is safe, or if you have to parse `{}` characters with a queue to understand your nesting level

Reply:
documented.